docs/language/informal/toplevel-inference.md

```diff
@@ -0,0 +1,338 @@
+﻿# Toplevel inference
+
+Owner: leafp@google.com
+
+Status: Proposal
+
+
+## Top level inference overview
+
+Top level inference has two phases:
+
+1. **Method override inference**
+    * If you omit a return type or parameter type from an overridden method,
```

Comment:
That's intended to be addressed in the section below:  

```
If there are multiple overridden/implemented methods and
the type to be filled in does not agree, it is an error.  
```

So your example is an inference failure error.